docker-compose.yaml

```diff
@@ -441,6 +441,12 @@ services:
     ports:
       - "8080:8080"
       - "9990:9990"
+    deploy:
+      resources:
+        limits:
+          # Set a maximum memory limit for the ZAC container for a more realistic
+          # scenario when testing ZAC locally or when running our integration tests.
+          memory: 4G
```

Comment:
We don't have max heap size set in the docker container, so ZAC will now use 1/4 = 1GB
To utilize this memory we can set the JVM initial and max heap sizes to 3500 to 3800MB in the Dockerfile **or** use 1.5GB here

Reply:
We should use the "relative" memory settings `-XX:InitialRAMPercentage`, `-XX:MaxRAMPercentage`, and `-XX:MinRAMPercentage` to ensure the JVM scales with the memory assigned to the container.

Reply:
This PR is not about changing the ZAC Docker container yet. It is just about limiting the amount of memory available to ZAC in our Docker Compose set up for more realistic local testing and itests. That is all. The real fix still needs to be done after this.

Reply:
That's why I approved, just wanted to mention that we should use relative instead of fixed size mem settings.